### PR TITLE
Allow chunks to use the persistent data API

### PR DESCRIPTION
**Pull request by brodybeckwith**

### CODE REVIEW

**Review by electronicboy — APPROVED** [review]

File: Spigot-Server-Patches/0444-Allow-chunks-to-use-the-persistent-data-API.patch [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+index 9cff8b88..49e63317 100644
+--- a/src/main/java/org/bukkit/craftbukkit/CraftChunk.java
++++ b/src/main/java/org/bukkit/craftbukkit/CraftChunk.java
+@@ -42,6 +42,11 @@ public class CraftChunk implements Chunk {
+     private static final DataPaletteBlock<IBlockData> emptyBlockIDs = new ChunkSection(0).getBlocks();
+     private static final byte[] emptyLight = new byte[2048];
+ 
++    // Paper start
++    private static final org.bukkit.craftbukkit.persistence.CraftPersistentDataTypeRegistry DATA_TYPE_REGISTRY = new org.bukkit.craftbukkit.persistence.CraftPersistentDataTypeRegistry();
++    public org.bukkit.craftbukkit.persistence.CraftPersistentDataContainer persistentDataContainer = new org.bukkit.craftbukkit.persistence.CraftPersistentDataContainer(DATA_TYPE_REGISTRY);
```
Comment (Spottedleaf) [inline Spigot-Server-Patches/0444-Allow-chunks-to-use-the-persistent-data-API.patch:61]: The persistent data container needs to be on the Chunk object (should be private final too). Don't need to deal with plugins that leak CraftChunk instances to also leak the persistent data.

Comment (electronicboy) [reply Spigot-Server-Patches/0444-Allow-chunks-to-use-the-persistent-data-API.patch:61]: This, actually, I forgot that people hold onto CraftChunk's \o/

**Review by electronicboy — CHANGES_REQUESTED** [review]
https://github.com/PaperMC/Paper/pull/2994/files/92291e23f64edc11a5e20e2e2ca3d27878fea4f6#r399564672

**Comment by sepiatonal** [comment]
Did @BrodyBeckwith drop this? If so I can submit a PR with the corrected change. This functionality would be useful for something I'm working on.

**Comment by bo3tools** [comment]
Any update on this? I would really love to utilize this feature in my project.

**Comment by aikar** [comment]
This is currently in review @ spigot and is showing traction, so rather us not add something that's about to immediately be a conflict.

So wait for it to finish at Spigot.

**Comment by MiniDigger** [comment]
https://hub.spigotmc.org/stash/projects/SPIGOT/repos/craftbukkit/pull-requests/672/overview

**Comment by lynxplay** [comment]
https://hub.spigotmc.org/stash/projects/SPIGOT/repos/craftbukkit/commits/77c894a2aba54cf2a0d9e8503fb97f874af68264

**Pull request closed without merging**